### DISCUSSION TITLE
Unauthorized error in peerjs server

### DISCUSSION THREAD

**Discussion by RamyaAshika** [peers/peerjs#1212] [Q&A] [answered]
I have hosted this peerjs server to heroku and I'm facing this error when I tried to connect

https://myapp.herokuapp.com/peerjs/6ba28c1d-8761-457e-ace4-31bcdbe4d32d/k1kpjuwftfr/id?i=0

and in my client code I have added

```ts
          const peer = new Peer({ 
          key: 'peerjs', 
          host: 'myapp.herokuapp.com', 
          secure: true, 
          port: 443, 
          generateClientId:id,
          config 
          });
```

instead of this

```
  peer = new Peer(id, {
              debug: 2,
              config
           });
```
and I have deployed my client code in apache web server and peerjs server is deployed in heroku. Can anyone please help me on this? Why its happening? @ afrokick  @ kidandcat or anyone else?

**Comment by afrokick** [comment]
What error you see?

**Comment by ASKNJ** [comment]
Facing the same error. Can someone help on this?

**Comment by ASKNJ** [comment]
> I have hosted this peerjs server to heroku and I'm facing this error when I tried to connect
> 
> https://myapp.herokuapp.com/peerjs/6ba28c1d-8761-457e-ace4-31bcdbe4d32d/k1kpjuwftfr/id?i=0
> 
> and in my client code I have added
> 
> ```
>       const peer = new Peer({ 
>       key: 'peerjs', 
>       host: 'myapp.herokuapp.com', 
>       secure: true, 
>       port: 443, 
>       generateClientId:id,
>       config 
>       });
> ```
> 
> instead of this
> 
> peer = new Peer(id, {
> debug: 2,
> config
> });
> and I have deployed my client code in apache web server and peerjs server is deployed in heroku. Can anyone please help me on this? Why its happening? @afrokick @kidandcat or anyone else?

Hey did you find anything on this?

**Comment by HanustaviraGA** [comment]
Still waiting for updates

**Comment by bosskabouter** [comment, marked as answer]
I don't think there is a generateClientId in client side options...  That's a serverside config option to have a custom id generator in case the default doesn't serve for clients connecting without id in the constructor.
